framework/include/fwk_core.h

```diff
@@ -142,6 +142,22 @@ int fwk_is_delayed_response_list_empty(fwk_id_t id, bool *is_empty);
  */
 int fwk_get_first_delayed_response(fwk_id_t id, struct fwk_event *event);
 
+/*!
+ * \brief Set the execution context to be used. An specific execution context
+ * can be associated to an element.
+ *
+ * \details When there are several transport channels that operate on
+ * unrelated resources, we can used dedicated execution context (event list ...)
+ * in order to parallelize the requests.
```

Comment:
This looks to me like we are introducing multi-threading again!
You can have an external application that is doing this parallel resource management. The firmware can be treated as one resource and protect this single resource by means of locking externally.

Reply:
The goal is to prevent a sequential access and associated delay for independent channel and resources. The context is only the linked list to use with a channel and doesn't touch the other channels or lists.

Reply:
This is probably the most controversial part that's why I put is at the end. The goal is to be able to run independent event lists on channel which access to private resources only

Reply:
We have only 1 Trusted application which is the entry point for linux or u-boot to invoke scmi request. Also the notification channel still need to stay connected with request one to send back notification